.github/workflows/test.yaml

```diff
@@ -661,10 +656,26 @@ jobs:
           echo "Collected labels: $LABELS"
       shell: bash
 
+    - name: Get changed files
+      id: get-changed-files
+      uses: tj-actions/changed-files@v34.5.1
+
+    - name: Detect changes files
+      id: detect-changes
+      run: |
+          CHANGED_FILES="${{ steps.get-changed-files.outputs.all_changed_files }}"
+          for file in CHANGED_FILES; do
+              if [[ "$file" == tests/nested/**/* ]] || [[ "$file" == tests/lib/nested.sh ]]; then
+                  echo "changes_found=true" >> $GITHUB_OUTPUT
+                  exit 0
+              fi
+          done
+          echo "changes_found=false" >> $GITHUB_OUTPUT
+
     - name: Run spread tests
       # run if the commit is pushed to the release/* branch or there is a 'Run
-      # nested' label set on the PR
-      if: "contains(steps.collect-pr-labels.outputs.labels, 'Run nested') || contains(github.ref, 'refs/heads/release/')"
+      # nested' label set on the PR or there is a change in a nested test/lib
+      if: "contains(steps.collect-pr-labels.outputs.labels, 'Run nested') || contains(steps.detect-changes.outputs.changes_found, 'true') || contains(github.ref, 'refs/heads/release/')"
```

Comment:
I think it's ok to do now. Later on we may find a reason where you don't want to run nested tests even though the files have changed, but we can always address it if someone complains

Reply:
well, perhaps to address this case we could have a "Skip nested" label